gateway/src/lib.rs

```diff
@@ -699,12 +713,24 @@ impl Actor {
         // to be committed. This doesn't apply to the root.
         // TODO: Once account abstraction is conveniently supported, there will be
         // no need for this initial funding of validators.
+        if network_name != *ROOTNET_ID {
+            if rt.curr_epoch() == 1 {
+                for v in validator_set.validators().iter() {
+                    rt.send(&v.addr, METHOD_SEND, None, INITIAL_VALIDATOR_FUNDS.clone())?;
+                }
+            }
 
-        if rt.curr_epoch() == 1 && network_name != *ROOTNET_ID {
-            for v in validator_set.validators().iter() {
-                rt.send(&v.addr, METHOD_SEND, None, INITIAL_VALIDATOR_FUNDS.clone())?;
+            // to prevent top-down checkpoint submissions from being stuck when the number of validators
+            // is too small (< 3), where if a new validator join and it doesn't have funds
+            // no new top-down checkpoints can be submitted, we add some balance also for
+            // non-initial validators.
+            if rt.curr_epoch() > 1 && validator_set.validators().len() < 4 {
```

Comment:
@cryptoAtwill, just to double-check with you. I don't think we should merge this one, as there is a really simple attack where before there are more than four validators, any validator could continuously join and leave to drain 1 FIL of the gateway at a time. I think for now let's track the bug here and figure out a better way to solve it. Tracking the issue here: https://github.com/consensus-shipyard/ipc-actors/issues/103